### DISCUSSION TITLE
Thoughts on scaling the UI?

### DISCUSSION THREAD

**Discussion by Joshfindit** [bee-san/pyWhat#92] [General]
As this tool gets more and more attention and more and more PRs are added, are there any choices that need to be made short or medium term to avoid predictable problems that could put off future new users?

As a ridiculous and contrived example:

`what 42`

Identified as:
- An age
- A house number
- An apartment number
- Studio 42 in Montreal
- The part number for a Lawn Boy gasket
- The number of items on the menu at Mirazur
- A shoe size (UK)
- The Ultimate Answer to Life, the Universe, and Everything
- ...
- (continuing on for hundreds or thousands of entries)

A new user might run something like this, scan through looking for something meaningful (which may be many scroll-lengths down), then give up before understanding the true value of `what`

**Comment by bee-san** [comment]
We have no docs on  it in the GitHub CLI, but we _do_ support filters:
https://github.com/bee-san/pyWhat/wiki/API#filters--distributions

So if the user wanted "Addresses" they could get back:

```
A house number
An apartment number
Studio 42 in Montreal
```

It _does_ require the user understanding _what_ they want though!

We also want to implement better checks (Luhn checks for debit cards). This isn't actually a github issue yet because my PC has been broken for 8 days and I only just fixed it lol 

I don't think there's much else we can do other than that? What do you think?

**Reply by Joshfindit** [reply]
Ok, that’s awesome.

I checked out the linked page and it looks like the future searching will still be feasible and manageable by using filters, even as `what` has tens of thousands of checks.

**Reply by bee-san** [reply]
Also for what it's worth, we start at rarity of 0.1 by default. 0.0 is for things that will spam the terminal, like `key:value` pairs. So if we wanted to add:

> an age

We have a way to add it without it showing up by default! :)
